src/components/Popups/TransactionPopup.tsx

```diff
@@ -1,46 +1,55 @@
+import Column from 'components/Column'
+import { ActivityRowStyledDescriptor } from 'components/WalletDropdown/MiniPortfolio/Activity'
+import { parseLocalActivity } from 'components/WalletDropdown/MiniPortfolio/Activity/parseLocal'
+import { PortfolioLogo } from 'components/WalletDropdown/MiniPortfolio/PortfolioLogo'
+import PortfolioRow from 'components/WalletDropdown/MiniPortfolio/PortfolioRow'
+import useENSName from 'hooks/useENSName'
+import { useCombinedActiveList } from 'state/lists/hooks'
 import { useTransaction } from 'state/transactions/hooks'
-import styled, { ThemeContext } from 'styled-components/macro'
-import { ExternalLink, ThemedText } from 'theme'
+import { ThemedText } from 'theme'
 import { ExplorerDataType, getExplorerLink } from 'utils/getExplorerLink'
 
-const RowNoFlex = styled(AutoRow)`
-  flex-wrap: nowrap;
-`
+import { PopupAlertTriangle } from './FailedNetworkSwitchPopup'
 
 export default function TransactionPopup({ hash }: { hash: string }) {
   const { chainId } = useWeb3React()
 
   const tx = useTransaction(hash)
-  const theme = useContext(ThemeContext)
+  const tokens = useCombinedActiveList()
+  const success = Boolean(tx?.receipt && tx?.receipt.status === 1)
 
-  if (!tx) return null
-  const success = Boolean(tx.receipt && tx.receipt.status === 1)
+  const activity = tx ? parseLocalActivity(tx, chainId ?? 0, tokens) : null
+  const { ENSName } = useENSName(activity?.otherAccount)
+  const explorerUrl = getExplorerLink(chainId ?? 0, hash, ExplorerDataType.TRANSACTION)
+
+  if (!chainId || !tx || !activity) return null
 
   return (
-    <RowNoFlex>
-      <div style={{ paddingRight: 16 }}>
-        {success ? (
-          <CheckCircle color={theme.accentSuccess} size={24} />
+    <PortfolioRow
+      left={
+        success ? (
+          <Column>
+            <PortfolioLogo
+              chainId={chainId}
+              currencies={activity.currencies}
+              images={activity.logos}
+              accountAddress={activity.otherAccount}
+            />
+          </Column>
         ) : (
-          <AlertCircle color={theme.accentFailure} size={24} />
-        )}
-      </div>
-      <AutoColumn gap="8px">
-        <ThemedText.BodyPrimary fontWeight={500}>
-          <TransactionSummary info={tx.info} />
-        </ThemedText.BodyPrimary>
-        {chainId && (
-          <ExternalLink href={getExplorerLink(chainId, hash, ExplorerDataType.TRANSACTION)}>
-            View on Explorer
-          </ExternalLink>
-        )}
-      </AutoColumn>
-    </RowNoFlex>
+          <PopupAlertTriangle />
+        )
+      }
+      title={<ThemedText.SubHeader fontWeight={500}>{activity.title}</ThemedText.SubHeader>}
+      descriptor={
+        <ActivityRowStyledDescriptor color="textSecondary">
+          {activity.descriptor}
+          {ENSName ?? activity.otherAccount}
+        </ActivityRowStyledDescriptor>
+      }
+      onClick={() => window.open(explorerUrl, '_blank')}
```

Comment:
ooof to `onClick` handlers being on divs - this isn't going to work for accessibility / keyboard tabbing unless you also manually define  tab indices and focus states

Reply:
you also aren't going to get link previews

Reply:
i think this is a problem with the MP row that needs to be addressed outside of this PR - also the tabbing matters more in the sidebar than in the toast probably.